### PR TITLE
[Feature] [Sql] Add Flink Jar sql

### LINKED ISSUE #1617: [Feature][task submit] support flink-sql submit jar&py mission
### Search before asking

- [X] I had searched in the [issues](https://github.com/DataLinkDC/dinky/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

You can extend the flink - SQL syntax, specify the jar or py files to start the session or application;(可以扩展flink-sql语法，指定 jar 或 py 文件启动session或application)

### Use case

 Such as: 
```sql
execute the run '< filepath >' < main - class > ';
```

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zackyoungh**
## Purpose of the pull request

1. Custom SQL Submit jar
> syntax:
``` sql
EXECUTE JAR WITH ( 
'uri'='...',
'main-class'='...',
'args'='...',-- 可选
'parallelism'='...',-- 可选
'savepoint-path'='...', -- 可选
'allow-non-restored-state'='false' -- 可选
)
```

> eg:
``` sql
EXECUTE JAR WITH ( 
'uri'='file:///C:/Users/ZackYoung/Downloads/paimon-flink-1.16-0.5-20230818.001833-127.jar',
'main-class'='org.apache.paimon.flink.action.FlinkActions',
'args'='mysql-sync-table --warehouse hdfs://10.8.16.157:9000/save --database cdc-test --table cdc_test1 --primary-keys id --mysql-conf hostname=121.5.136.161 --mysql-conf port=3371 --mysql-conf username=root --mysql-conf password=dinky --mysql-conf database-name=cdc-test --mysql-conf table-name=table_1 --mysql-conf server-time-zone=Asia/Shanghai --table-conf bucket=4 --table-conf changelog-producer=input --table-conf sink.parallelism=1',
'parallelism'='',
'savepoint-path'='' 
)
```
issue #1617

### CODE REVIEW

File: dinky-core/src/main/java/org/dinky/explainer/Explainer.java [hunk 1]
```diff
@@ -277,7 +275,6 @@ record = new SqlExplainResult();
                 } else {
                     executor.executeSql(item.getValue());
                 }
-                record.setType("DATASTREAM");
```
Comment (leechor) [inline dinky-core/src/main/java/org/dinky/explainer/Explainer.java:280]: does here affect batch model?

File: dinky-core/src/main/java/org/dinky/job/JobManager.java [hunk 2]
```diff
@@ -569,19 +569,14 @@ public JobResult executeSql(String statement) {
                 } else {
                     for (StatementParam item : jobParam.getExecute()) {
                         executor.executeSql(item.getValue());
-                        if (!useStatementSet) {
-                            break;
```
Comment (leechor) [inline dinky-core/src/main/java/org/dinky/job/JobManager.java:573]: assure this logic @aiwenmo

File: dinky-client/dinky-client-1.16/src/main/java/org/dinky/executor/CustomTableEnvironmentImpl.java [hunk 3]
```diff
@@ -180,6 +180,8 @@ public SqlExplainResult explainSqlRecord(String statement, ExplainDetail... extr
             record.setType("Explain DML");
         } else if (operation instanceof QueryOperation) {
             record.setType("Query DML");
+        } else if (operation instanceof QueryOperation) {
+            record.setType("Query DML");
         } else {
```
Comment (aiwenmo) [inline dinky-client/dinky-client-1.16/src/main/java/org/dinky/executor/CustomTableEnvironmentImpl.java:185]: Is the code duplicated?

File: dinky-executor/src/test/java/org/dinky/trans/ExecuteJarParseStrategyTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import cn.hutool.core.lang.Assert;
+
+class ExecuteJarParseStrategyTest {
+
+    @Test
+    void getInfo() {
+        String statement = "EXECUTE JAR WITH ( \n"
+                + "'uri'='file:///C:/Users/ZackYoung/Downloads/paimon-flink-1.16-0.5-20230818.001833-127.jar',\n"
+                + "'main-class'='org.apache.paimon.flink.action.FlinkActions',\n"
+                + "'args'='mysql-sync-table --warehouse hdfs://10.8.16.157:9000/save --database cdc-test --table cdc_test1 --primary-keys id --mysql-conf hostname=121.5.136.161 --mysql-conf port=3371 --mysql-conf username=root --mysql-conf password=dinky --mysql-conf database-name=cdc-test --mysql-conf table-name=table_1 --mysql-conf server-time-zone=Asia/Shanghai --table-conf bucket=4 --table-conf changelog-producer=input --table-conf sink.parallelism=1',\n"
```
Comment (aiwenmo) [inline dinky-executor/src/test/java/org/dinky/trans/ExecuteJarParseStrategyTest.java:36]: Please use 127.0.0.1

**Review by aiwenmo — CHANGES_REQUESTED** [review]

File: dinky-core/src/main/java/org/dinky/job/JobManager.java (hunk 2 above)
Comment (aiwenmo) [reply dinky-core/src/main/java/org/dinky/job/JobManager.java:573]: Restore first, I'm refactoring.

**Pull request closed without merging**